liquibase: add support for Amazon Redshift

sir4ur0n opened the pull request:
- [x] Prerequisite: https://github.com/NixOS/nixpkgs/pull/168177 is merged

The Liquibase package already provided PostgreSQL and MySQL support, this commit
also adds support for Amazon Redshift.

As per Liquibase Redshift documentation, 2 JARs must be added to the classpath:
* Redshift JDBC
* Liquibase Redshift extension JAR

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Without this, using Nixpkgs `liquibase` to interact with a Redshift database is inconvenient: manually download the JARs, copy and update the `liquibase` script to add those to the classpath, etc.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

sir4ur0n commented:
This has been open for a month now and nobody reviewed it. Is there anything I can do to move this forward please? 😅

bennyandresen reviewed (approved):
Result of `nixpkgs-review pr 150966` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>liquibase</li>
    <li>liquibase_redshift_extension</li>
    <li>redshift_jdbc</li>
  </ul>
</details>

Builds and license checked. LGTM!

/status needs_merger

timokau reviewed (changes requested):
Looks good! For consistency with our [contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), could you split this into 3 commits?

- `redshift_jdbc: init at 2.1.0.3`
- `liguibase-redshift-extension: init at 4.8.0`
- `liquibase: add support for Amazon Redshift`

Unfortunately liquibase does not have a maintainer right now. Would you be willing to take over maintenance? Until then, maybe @roberth has an opinion on this (based on the commit history).

timokau commented on pkgs/development/java-modules/redshift_jdbc/default.nix:
```diff
@@ -0,0 +1,28 @@
+  meta = with lib; {
+    homepage = "https://github.com/aws/amazon-redshift-jdbc-driver/";
+    description =
+      "JDBC driver for Amazon Redshift allowing Java programs to connect to a Redshift database";
```
```suggestion
      "JDBC 4.2 driver for Amazon Redshift allowing Java programs to connect to a Redshift database";
```

sir4ur0n replied:
Addressed in the latest version

timokau commented:
Hi!

The marvin-mk2 bot is now [discontinued](https://github.com/timokau/marvin-mk2/issues/34#issuecomment-1100656280). I have removed the relevant tags from this PR. If you still need someone to look at it, one option would be to ask in [this discourse thread](https://discourse.nixos.org/t/prs-ready-for-review/3032).

I am posting this notice to all open PRs with the `marvin` tag. Please understand that following all of these discussions would take too much time, so I will unsubscribe from this PR unless I have already been involved in it before this message.

roberth commented:
I don't use it and I don't recall contributing to this package, but it looks fine, apart from using `_` in the attribute name instead of `-`. `-` is encouraged in packages names and attribute names, except where it indicates a variant, like `_headless` or a version class like `_unstable` or `_3`. This hasn't been applied consistently before, fwiw.

Commit:
redshift_jdbc: init at 2.1.0.3

Commit:
liquibase_redshift_extension: init at 4.8.0

Commit:
liquibase: add support for Amazon Redshift

The Liquibase package already provided PostgreSQL and MySQL support, this commit
also adds support for Amazon Redshift.

As per Liquibase Redshift documentation, 2 JARs must be added to the classpath:
* Redshift JDBC
* Liquibase Redshift extension JAR

sir4ur0n commented:
I just force-pushed to this branch:
* @timokau I split in 3 commits as you suggested, I agree it's cleaner
* added `4.2` in the description as suggested

@timokau I don't feel confident being maintainer of `liquibase` package because I don't understand most of what's going on in this package :sweat_smile: Also I don't use it much

@roberth I did it that way for consistency with other `_jdbc` packages, in my opinion consistency in a sub-part of the codebase is more important, but if it's blocking for merge I will change it (only for the 2 new packages as it would be a breaking change for others I think), please let me know

timokau reviewed (approved):
Thanks! I think there's good arguments for both sides of the `_` vs `-` issue. It would probably be "cleanest" to fix the other packages as well, but that would break things unnecessarily. I'll merge this as-is, I hope you don't mind @roberth.
